### PR TITLE
[4.0] GD, iconv and intl settings test

### PR DESCRIPTION
**Pull request by brianteeman**
In a recent PR the status of these two php functions was added to the system info.

This PR adds the same information to the pre-install and pre-update checks

And additionally adds GD to all three places

In order to test this page you will need to force joomla to see an update. One way to do this is to set the update channel to custom and use this url https://update.joomla.org/core/nightlies/next_major_list.xml

When you have done that then you will see the pre-update page that this PR is addressing.
image

### CODE REVIEW

**Comment by richard67** [comment]
@brianteeman Could you fix the code style errors reported here? https://ci.joomla.org/joomla/joomla-cms/43947/1/6 Thanks in advance.

**Comment by brianteeman** [comment]
apologies for the delay is was speaking at a conference

File: installation/language/en-GB/joomla.ini [hunk 1]
```diff
@@ -196,6 +196,7 @@ INSTL_DISPLAY_ERRORS="Display Errors"
 INSTL_ERROR="Error"
 INSTL_ERROR_DB="Some errors occurred while populating the database: %s."
 INSTL_ERROR_INITIALISE_SCHEMA="Can't initialise database schema."
+INSTL_EXTENSION_AVAILABLE="Available"
```
Comment (Quy) [inline installation/language/en-GB/joomla.ini]:
```suggestion
INSTL_EXTENSION_AVAILABLE="%s Available"
```

**Comment by PhilETaylor** [comment]
Just a thought - maybe we should also add the GD check?

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on bcb92b5459cf94c127b8d373952d941f407d4ea4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33977">issues.joomla.org/tracker/joomla-cms/33977</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on bcb92b5459cf94c127b8d373952d941f407d4ea4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33977">issues.joomla.org/tracker/joomla-cms/33977</a>.</sub>

File: installation/src/Model/ChecksModel.php [hunk 2]
```diff
@@ -204,6 +204,20 @@ public function getPhpSettings()
...
+		// Check for iconv support
+		$setting = new \stdClass;
+		$setting->label = Text::sprintf('INSTL_EXTENSION_AVAILABLE', 'iconv');
+		$setting->state = function_exists('iconv');
+		$setting->recommended = true;
+		$settings[] = $setting;
+
+		// Check for intl support
+		$setting = new \stdClass;
+		$setting->label = Text::sprintf('INSTL_EXTENSION_AVAILABLE', 'instl');
```
Comment (infograf768) [inline installation/src/Model/ChecksModel.php:223]: `intl`, not `instl`

Comment (richard67) [reply installation/src/Model/ChecksModel.php:223]: Correct. The reason for this typo might be the `INSTL_` at the beginning of the language string constants, which stands for "installer". Here of course it has to be `intl` for the library name.

**Comment by brianteeman** [comment]
Its only a language string change so I guess with a maintainer review to confirm the other tests are still valid

**Comment by richard67** [comment]
> ... so I guess with a maintainer review to confirm the other tests are still valid

Am on the way. How about Phil's idea to do that also for the GD library check? You wanna make another PR?

**Comment by brianteeman** [comment]
yes - i will do a separate pr - out of time right now

**Comment by richard67** [comment]
I've restored the previous test results in the issue tracker so they are properly counted again, because the commit which invalidated them was just a text change.

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33977">issues.joomla.org/tracker/joomla-cms/33977</a>.</sub>

**Comment by brianteeman** [comment]
As it hadnt been merged yet I added GD support to this PR to avoid any merge conflicts etc.

Please retest

**Comment by richard67** [comment]
Back to pending.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33977">issues.joomla.org/tracker/joomla-cms/33977</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 134ec3189c592f1395f9d833a5df7639ad398b86

#### Pre-Update Check:

image

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33977">issues.joomla.org/tracker/joomla-cms/33977</a>.</sub>

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 134ec3189c592f1395f9d833a5df7639ad398b86<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33977">issues.joomla.org/tracker/joomla-cms/33977</a>.</sub>

**Comment by infograf768** [comment]
rtc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33977">issues.joomla.org/tracker/joomla-cms/33977</a>.</sub>

**Comment by richard67** [comment]
Thanks!

**Comment by brianteeman** [comment]
Thx

**Pull request merged**